fix: 푸시 알림 전송 중 FcmTopic NPE 발생

# 📝 작업 내용

- 모임에 참여 후 아래 에러 발생

```
2024-09-19 16:06:07.732 [ERROR] [scheduling-1] [] [o.s.s.s.TaskUtils$LoggingErrorHandler] - Unexpected error occurred in scheduled task
java.lang.NullPointerException: Cannot invoke "com.ody.notification.domain.FcmTopic.getValue()" because the return value of "com.ody.notification.domain.Notification.getFcmTopic()" is null
	at com.ody.notification.domain.message.GroupMessage.<init>(GroupMessage.java:17)
	at com.ody.notification.service.FcmPushSender.sendPushNotification(FcmPushSender.java:27)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:354)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:196)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:768)
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:392)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:768)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:720)
	at com.ody.notification.service.FcmPushSender$$SpringCGLIB$$0.sendPushNotification(<generated>)
	at com.ody.notification.service.NotificationService.lambda$scheduleNotification$0(NotificationService.java:77)
	at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

--- a/backend/src/main/java/com/ody/notification/domain/Notification.java
+++ b/backend/src/main/java/com/ody/notification/domain/Notification.java
@@ -59,13 +59,13 @@ public class Notification extends BaseEntity {
         this(null, mate, type, sendAt, status, fcmTopic);
     }
 
-    public static Notification createEntry(Mate mate) {
+    public static Notification createEntry(Mate mate, FcmTopic fcmTopic) {
         return new Notification(
                 mate,
                 NotificationType.ENTRY,
                 LocalDateTime.now(),
                 NotificationStatus.PENDING,
-                null
+                fcmTopic
         );
     }
 

--- a/backend/src/main/java/com/ody/notification/service/NotificationService.java
+++ b/backend/src/main/java/com/ody/notification/service/NotificationService.java
@@ -41,14 +41,14 @@ public class NotificationService {
 
     @Transactional
     public void saveAndSendNotifications(Meeting meeting, Mate mate, DeviceToken deviceToken) {
-        saveAndSendEntryNotification(mate);
         FcmTopic fcmTopic = new FcmTopic(meeting);
+        saveAndSendEntryNotification(mate, fcmTopic);
         fcmSubscriber.subscribeTopic(fcmTopic, deviceToken);
         saveAndSendDepartureReminderNotification(meeting, mate, fcmTopic);
     }
 
-    private void saveAndSendEntryNotification(Mate mate) {
-        Notification notification = Notification.createEntry(mate);
+    private void saveAndSendEntryNotification(Mate mate, FcmTopic fcmTopic) {
+        Notification notification = Notification.createEntry(mate, fcmTopic);
         saveAndSendNotification(notification);
     }
 

--- a/backend/src/main/java/com/ody/notification/service/NotificationService.java
+++ b/backend/src/main/java/com/ody/notification/service/NotificationService.java
@@ -75,7 +75,12 @@ public class NotificationService {
     public void scheduleNotification(FcmGroupSendRequest fcmGroupSendRequest) {
         Instant startTime = fcmGroupSendRequest.notification().getSendAt().toInstant(KST_OFFSET);
         taskScheduler.schedule(() -> fcmPushSender.sendPushNotification(fcmGroupSendRequest), startTime);
-        log.info("{} 상태 알림 {}에 스케줄링 예약", fcmGroupSendRequest.notification().getStatus(), startTime);
+        log.info(
+                "{} 타입 {} 상태 알림 {}에 스케줄링 예약",
+                fcmGroupSendRequest.notification().getType(),
+                fcmGroupSendRequest.notification().getStatus(),
+                startTime
+        );
     }
 
     @EventListener(ApplicationReadyEvent.class)

--- a/backend/src/main/java/com/ody/route/service/OdsayRouteClient.java
+++ b/backend/src/main/java/com/ody/route/service/OdsayRouteClient.java
@@ -37,6 +37,7 @@ public class OdsayRouteClient implements RouteClient {
                 .uri(makeURI(origin, target))
                 .retrieve()
                 .body(OdsayResponse.class);
+        log.info("ODsay API 호출 : {}", response);
         return Objects.requireNonNullElseGet(response, () -> {
             throw new OdyServerErrorException("서버 에러");
         });

